.clang-format

```diff
@@ -0,0 +1,200 @@
+---
+Language:        Cpp
+# BasedOnStyle:  Google
```

Comment:
Generated? Could you share how you gain this config file?

Reply:
1. `cd` to a non-project directory
2. `clang-format-12 -style=google -dump-config > .clang-format`
3. change `ColumnLimit` to 120

Reply:
I'd prefer `ColumnLimit` to 200...Nowadays display device is far more wider than 80 or 120 already, and wrap function calls or string literals are really clumsy.

Reply:
The old limit is 120, so I just inherit it. I think maybe 200 is not friendly to my small 1080p laptop 🤣 , but I am open for it. What do other guys think about it? cc @git-hulk @torwig @caipengbo @ShooterIT

Reply:
I also think 120 is enough, we should rarely have too long lines?

Reply:
I also think 120 is enough.